misc/clock.ru

```diff
@@ -71,16 +84,34 @@ loop do
   # and the data to publish.
 
   # The Redis approach.
+  # 
+  # I have a preference for this one just because that's what I did in the past,
+  # (I used 'channel' and 'pattern' as the two names)...
+  # ...but we can surely use the 'subject' name if you feel it's better. I just
+  # think it might be a bit confusing where pattern matching is concerned.
```

Comment:
I was hoping to not have a name but just two arguments, a thing that identifies the subject, channel, or whatever and the message itself. With just two argument it seems like overkill to use a hash.

Reply:
This goes back to the question of WebSocket op-codes for outgoing messages.

To support WebSocket messages we might need 3 arguments:

```ruby
def subscribe(subject, is_pattern = false, as_text = true)
end
```

Because we have a number of default values, I think "named arguments" (which internally use a small Hash object) might be more convenient?

If it was only 2 arguments, I would totally agree. I don't think it's a big issue... but with 3 arguments, I don't know. What do you think?

Reply:
See previous comment about op-codes, strings, and binary. Might be worth thinking about some more.